### PR TITLE
Mail listener filters fix

### PR DESCRIPTION
**Pull request by MosheEichler**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/42996

## Description
Added a `HEADER` to the query search to support office365 IMAP servers

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/MailListener/ReleaseNotes/1_0_7.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Mail Listener v2
+- Fixed an issue where ***fetch-incidents*** with filters did not return emails from the office365 IMAP server.
```
Comment (amshamah419) [inline Packs/MailListener/ReleaseNotes/1_0_7.md:4]:
```suggestion
- Fixed an issue where ***fetch-incidents*** with filters would not return emails from the Office365 IMAP server.
```

**Review by reutshal — CHANGES_REQUESTED** [review]
Lets not change this for everyone. 
I think we should expose this as an integration parameter and build the query according to it.
It should be added as not mandatory and the default will be the old behavior.
We can add additional info to this parameter explaining in which case we want to use it.

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.yml [hunk 2]
```diff
@@ -37,6 +37,15 @@ configuration:
   name: permittedFromDomain
   required: false
   type: 0
+- display: Mail service
+  additionalinfo: There are changes between email services regarding how to set the query with filters (such as the sender's email address) to fetch emails
+    So if you are having anhy issues with the fetch emails when you add filters consider using the Office365 email service
```
Comment (MosheEichler) [inline Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.yml:42]: @amshamah419 Please take a look
Thanks!

Comment (amshamah419) [inline Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.yml:42]:
```suggestion
  additionalinfo: There are differences between email services regarding the format of the query with filters while fetching emails (such as a sender's email).
    If you are experiencing any issues fetching emails when you add filters, consider using the Office365 email service.
```

Comment (reutshal) [reply Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.yml:42]: @MosheEichler Please accept the suggestion

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2_test.py [hunk 3]
```diff
@@ -109,8 +110,61 @@ def test_convert_to_incident():
                 '4:*'
             ]
         ),
+        (
+            datetime(year=2020, month=10, day=1),
+            'SomeIMAPServer',
```
Comment (reutshal) [inline Packs/MailListener/Integrations/MailListenerV2/MailListenerV2_test.py:115]: Why do we need this example? When will we get "SomeIMAPServer"?

**Review by reutshal — CHANGES_REQUESTED** [review]

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2_test.py (hunk 3 above)
Comment (MosheEichler) [reply Packs/MailListener/Integrations/MailListenerV2/MailListenerV2_test.py:115]: It is unnecessary.

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py [hunk 4]
```diff
@@ -294,24 +301,30 @@ def generate_search_query(time_to_fetch_from: Optional[datetime],
...
     Input example:
     time_to_fetch_from: datetime.datetime(2020, 8, 7, 12, 14, 32, 918634, tzinfo=datetime.timezone.utc)
+    server_type: 'Office365'
     permitted_from_addresses: ['test1@mail.com', 'test2@mail.com']
     permitted_from_domains: ['test1.com', 'domain2.com']
     output example:
     ['OR',
      'OR',
      'OR',
+     'HEADER', # For Office365 IMAP server only
```
[the remark this answers is not in the document]
Comment (MosheEichler) [reply Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py]: > add one example without the header

And check if the UT fails?

**Comment by reutshal** [comment]
@Itay4 Can you please take a quick look? Thanks

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py [hunk 5]
```diff
@@ -326,6 +344,8 @@ def generate_search_query(time_to_fetch_from: Optional[datetime],
         messages_query = OR(from_=permitted_from_addresses_list + permitted_from_domains_list).format()
         # Removing Parenthesis and quotes
         messages_query = messages_query.strip('()').replace('"', '')
+        if mail_service == 'Office365':
+            messages_query = messages_query.replace('FROM', 'HEADER FROM')
```
Comment (Itay4) [inline Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py:348]: i dont really like this solution, adding an integration param seems like an overkill

can we query with both `FROM` and `HEADER FROM` ?

Comment (MosheEichler) [reply Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py:348]: In the mail service we checked with (Gmail) it is possible but to keep BC (and to ensure work with all mail services) we preferred to add a parameter
@reutshal FYI

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py [hunk 6]
```diff
@@ -187,6 +188,7 @@ def fetch_incidents(client: IMAPClient,
         last_run: The greatest incident created_time we fetched from last fetch
         first_fetch_time: If last_run is None then fetch all incidents since first_fetch_time
         include_raw_body: Whether to include the raw body of the mail in the incident's body
+        with_header: Whether to add the word HEADER to the query search
```
Comment (reutshal) [inline Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py:191]:
```suggestion
        with_headers: Whether to add headers to the search query
```

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py [hunk 7]
```diff
@@ -162,6 +162,7 @@ def fetch_incidents(client: IMAPClient,
                     last_run: dict,
                     first_fetch_time: str,
                     include_raw_body: bool,
+                    with_header: bool,
```
Comment (reutshal) [inline Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py:165]:
```suggestion
                    with_headers: bool,
```

Comment (reutshal) [reply Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py:165]: change in all places please

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py [hunk 8]
```diff
@@ -235,6 +239,7 @@ def fetch_mails(client: IMAPClient,
         client: IMAP client
         time_to_fetch_from: Fetch all incidents since first_fetch_time
         include_raw_body: Whether to include the raw body of the mail in the incident's body
+        with_header: Whether to add the word HEADER to the query search
```
Comment (reutshal) [inline Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py:242]:
```suggestion
        with_headers: Whether to add headers to the search query
```

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.yml [hunk 9]
```diff
@@ -37,6 +37,12 @@ configuration:
   name: permittedFromDomain
   required: false
   type: 0
+- defaultvalue: 'false'
+  additionalinfo: Whether to add the word HEADER to the query search when filtering by senders email address, In some of the mails services it's required.
+  display: Include HEADER in the query seracrh
```
Comment (reutshal) [inline Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.yml:42]:
```suggestion
  display: Query search with headers
```

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.yml [hunk 10]
```diff
@@ -37,6 +37,12 @@ configuration:
   name: permittedFromDomain
   required: false
   type: 0
+- defaultvalue: 'false'
+  additionalinfo: Whether to add the word HEADER to the query search when filtering by senders email address, In some of the mails services it's required.
```
Comment (reutshal) [inline Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.yml:41]:
```suggestion
  additionalinfo: Whether to query search with headers when filtering by sender's email address.
```

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.yml [hunk 11]
```diff
@@ -37,6 +37,12 @@ configuration:
...
+  display: Include HEADER in the query seracrh
+  name: with_header
```
Comment (reutshal) [inline Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.yml:43]:
```suggestion
  name: with_headers
```

File: Packs/MailListener/ReleaseNotes/1_0_7.md [hunk 12]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Mail Listener v2
+- Fixed an issue where ***fetch-incidents*** with filters would not return emails from the Office365 mail service.
```
Comment (reutshal) [inline Packs/MailListener/ReleaseNotes/1_0_7.md:4]:
```suggestion
- Fixed an issue where ***fetch-incidents*** with filters would not return emails from some of the mail services.
```

**Review by reutshal — CHANGES_REQUESTED** [review]

**Review by reutshal — APPROVED** [review]

**Pull request merged**